docs/source/development/package_releases.md

```diff
@@ -0,0 +1,163 @@
+---
+jupytext:
+  formats: md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.13.8
+kernelspec:
+  display_name: vr_python3
+  language: python
+  name: vr_python3
+---
+
+# Package release process
+
+The package release process has three stages:
+
+* Merge changes from the `develop` branch onto `main` that will form the new release
+  version.
+* Publish a release on GitHub - this is basically just a specific tagged commit on
+  `main` that has some associated release notes.
+* Publish the built code packages to PyPI - this is the packaged version of the code
+  that users will install and use. The `virtual_ecosystem` uses the trusted publishing
+  mechanism to make it easy to add new release to PyPI.
+
+## Generate the code commit to be released
+
+The release process for new versions of the `virtual_ecosystem` package is managed using
+pull requests to the `main` branch to create a specific commit that will be released.
+The steps of the process are:
+
+1. **Create a new release branch** from the `develop` branch called `release/X.Y.Z` ,
+   where `X.Y.Z` is the expected release version number.
+
+1. Update the `pyproject.toml` file to use the expected release versions number and
+   commit that change. You can use `poetry version` command to increment the major,
+   minor and patch version but it is almost as easy to edit the file by hand.
+
+1. **Start a pull request against the `main` branch**. The PR will transfer all of the
+   changes to the `develop` branch since the last release on to the `main` branch. The
+   PR description should provide a good explanation of the functionality that is being
+   changed or added in this version, and an explanation of the suggested version number
+   increment. For example, "This PR fixes a bug in calculating plant growth and so is a
+   patch release from v.0.1.8 to v0.1.9".
+
+1. **The CI testing obviously now needs to pass**. Any issues need to be resolved by
+   commits or PRs onto the `release/x.y.x` branch.
+
+1. **The PR also must be reviewed**. The code itself has already gone through the
+   review process to be merged into `develop`, so this is not a code review so much as a
+   review of the justification for a release.
+
+1. **The branch can then be merged into `main`**. Do _not_ delete the release branch at
+   this point.
+
+1. **Create a second PR to merge the release branch into `develop`.** This is to
+   synchronise any release changes (including the version number change) between the
+   `main` and `develop` branches.
```

Comment:
Is the addition of new stuff to the develop branch still going on during the release period (the steps above). I wonder if it will be cleaner to just rebase the develop branch with the main branch.

Reply:
@dalonsoa Yes - the intention is that everything that happens on the release branch ends up on both `main` and `develop`. We'd still need a PR to do a rebase? I have to admit I don't fully understand how rebase differs here!

Reply:
Either way, lets stick with this description for now.